class.jetpack-jitm.php

```diff
@@ -233,14 +233,10 @@ static function get_messages( $message_path, $query ) {
 
 		// attempt to get from cache
 		$envelopes  = get_transient( 'jetpack_jitm_' . $path );
-		$from_cache = false;
 
-		// if something is in the cache and it was put in the cache after the last heartbeat or sync, use it
-		$last_heartbeat = Jetpack_Options::get_option( 'last_heartbeat', time() );
-		$last_sync      = Jetpack_Options::get_option( 'last_sync', time() );
-		if ( $envelopes && $last_heartbeat < $envelopes['response_time'] || $last_sync < $envelopes['response_time'] ) {
-			$from_cache = true;
-		}
+		// if something is in the cache and it was put in the cache after the last sync we care about, use it
+		$last_sync = (int) get_transient( 'jetpack_last_sync' );
+		$from_cache = $envelopes && $last_sync < $envelopes[ 'response_time' ];
```

Comment:
Ah! I see, of course - your JITM cache is almost always less than an hour, so my comment is moot.